cmd/bootstrap/cmd/key.go

```diff
@@ -9,47 +9,108 @@ import (
+	keyCmd.Flags().StringVar(&flagRole, "role", "", "node role (can be \"collection\", \"consensus\", \"execution\", \"verification\" or \"access\")")
+	_ = keyCmd.MarkFlagRequired("role")
+	keyCmd.Flags().StringVar(&flagAddress, "address", "", "network address")
+	_ = keyCmd.MarkFlagRequired("address")
+
+	keyCmd.Flags().BoolVar(&flagNetworkKey, "networking", false, "generate networking key only")
+	keyCmd.Flags().BoolVar(&flagNetworkKey, "staking", false, "generate staking key only")
+	keyCmd.Flags().BoolVar(&flagNetworkKey, "machine", false, "generate machine key only")
+
+	keyCmd.Flags().BytesHexVar(&flagNetworkSeed, "networking-seed", generateRandomSeed(), fmt.Sprintf("hex encoded networking seed (min %v bytes)", minSeedBytes))
+	keyCmd.Flags().BytesHexVar(&flagStakingSeed, "staking-seed", generateRandomSeed(), fmt.Sprintf("hex encoded staking seed (min %v bytes)", minSeedBytes))
+	keyCmd.Flags().BytesHexVar(&flagMachineSeed, "machine-seed", generateRandomSeed(), fmt.Sprintf("hex encoded machine account seed (min %v bytes)", minSeedBytes))
+}
+
 // keyCmdRun generate the node staking key, networking key and node information
 func keyCmdRun(_ *cobra.Command, _ []string) {
 	// validate inputs
 	role := validateRole(flagRole)
 	validateAddressFormat(flagAddress)
-	networkSeed := validateSeed(flagNetworkSeed)
-	stakingSeed := validateSeed(flagStakingSeed)
 
-	log.Debug().Msg("will generate networking key")
-	networkKeys, err := run.GenerateNetworkingKeys(1, [][]byte{networkSeed})
-	if err != nil {
-		log.Fatal().Err(err).Msg("cannot generate networking key")
+	var networkKey crypto.PrivateKey
+	var stakingKey crypto.PrivateKey
+	var machineKey crypto.PrivateKey
+
+	if !flagNetworkKey && !flagStakingKey && !flagMachineKey {
+		flagNetworkKey = true
+		flagStakingKey = true
+		flagMachineKey = true
 	}
-	log.Info().Msg("generated networking key")
 
-	log.Debug().Msg("will generate staking key")
-	stakingKeys, err := run.GenerateStakingKeys(1, [][]byte{stakingSeed})
-	if err != nil {
-		log.Fatal().Err(err).Msg("cannot generate staking key")
+	if flagNetworkKey {
+		log.Debug().Msg("will generate networking key")
+		networkSeed := validateSeed(flagNetworkSeed)
+		networkKeys, err := run.GenerateNetworkingKeys(1, [][]byte{networkSeed})
+		if err != nil {
+			log.Fatal().Err(err).Msg("cannot generate networking key")
+		}
+		networkKey = networkKeys[0]
+		log.Info().Msg("generated networking key")
+	}
+
+	if flagStakingKey {
+		log.Debug().Msg("will generate staking key")
+		stakingSeed := validateSeed(flagStakingSeed)
+		stakingKeys, err := run.GenerateStakingKeys(1, [][]byte{stakingSeed})
+		if err != nil {
+			log.Fatal().Err(err).Msg("cannot generate staking key")
+		}
+		stakingKey = stakingKeys[0]
+		log.Info().Msg("generated staking key")
+	}
+
+	if flagMachineKey {
+		log.Debug().Msg("will generate machine account key")
+		machineSeed := validateSeed(flagMachineSeed)
+		machineKeys, err := run.GenerateNetworkingKeys(1, [][]byte{machineSeed})
```

Comment:
It's a bit confusing to use `GenerateNetworkingKey` for the machine account key. We could change the method to `GenerateECDSAKey` and use it for both networking and machine account key, or we could add an alias method `GenerateMachineAccountKey`.